### DISCUSSION TITLE
CI Issues

### DISCUSSION THREAD

**Discussion by cullzie** [aapatre/Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE#172] [Ideas]
Creating this discussion to track any issues with CI

**Comment by fakeid30** [comment]
Good idea. I am debating whether to post the dummy account for CI @aapatre setup here.

**Reply by cullzie** [reply]
Ye we might need to remove the udemy login part for now. I'll try out a few possible fixes but it might be more effort to maintain than it is worth

**Comment by cullzie** [comment]
Headless browser seems to be always triggering robot check at the moment.
If I can't find a solution I would suggest we remove this from the CI run for the moment. 
Instead just add a sanity check for the scrapers and expand unittests.

**Reply by fakeid30** [reply]
Yeah that can be done.

**Reply by fakeid30** [reply]
Another idea is to run the CI suite from GitLab while hosting the code in GitHub (it's doable)

**Reply by cullzie** [reply]
Ye unittests and some smoke/sanity tests for the scrapers is straightforward in github. 
It might be possible to setup selenium grid on gitlab. I haven't looked at the options over there yet

**Reply by fakeid30** [reply]
@cullzie CI is now running without the --debug flag.

**Reply by cullzie** [reply]
Ye I added in accept-language header #174 which seemed to fix it locally but it was still having some issues just after I merged it. At least it's back on the right track

**Reply by fakeid30** [reply]
the issue was happening because udemy locked out the account. Needed to reset the pass and update GH secret.

**Comment by fakeid30** [comment]
apparently udemy has done something with our dummy account, i was reviewing CI logs and it's failing to login for a LONG time. I'll ask @aapatre to change it again.

**Comment by cullzie** [comment]
I think we should split up the checks we are running at the moment since the login is no longer working via headless browser and it has always been quite a flaky test.

1. Have smoke test for each site we scrape to determine that it still works.
2. Have unit tests to test code is working as we expect

For udemy logic we'll just need to rely on users creating bugs if/when they see issues

@fakeid30 What do you think?

**Reply by fakeid30** [reply]
Yep seems to be the best way for now.

**Comment by fakeid30** [comment]
@cullzie homebrew 2.7.0 has changed and now it's breaking mac installing. We need to fix the mac syntax for CI.

**Reply by fakeid30** [reply]
Fixed.

**Reply by fakeid30** [reply]
@cullzie CI still failing.

**Comment by cullzie** [comment]
@fakeid30 Need to rework the CI jobs a bit. Can probably just set it up to run on ubuntu now. 
Might need @aapatre to have a look at the account currently being used. Looks like it is having issues logging in
